src/Saml2/Provider.php

```diff
@@ -218,7 +219,7 @@ public function getServiceProviderEntityDescriptor(): EntityDescriptor
     {
         $entityDescriptor = new EntityDescriptor();
         $entityDescriptor
-            ->setEntityID(URL::to('auth/saml2'))
+            ->setEntityID($this->getConfig('sp_entityid', URL::to('auth/saml2')))
```

Comment:
Hmm, what does `URL::to('auth/saml2')` produce here? 
Should the entity id be inside the url to?

Reply:
Good point, I should probably change the [documentation](https://github.com/SocialiteProviders/Providers/pull/707/files#diff-a3aac639db6923f75aefb2cf21a84fc924dba38f3bb97659cf1748b2ffc41518R158) to look like a URI
```
'saml2' => [
  'metadata' => 'https://idp.co/metadata/xml',
  'sp_entityid' => 'https://my.domain.com/my/custom/entityid',
],
```

from the SAML2 Specification
> ... The syntax of such an identifier is a URI of not more than 1024 characters in length. It is RECOMMENDED that a system entity use a URL containing its own domain name to identify itself.


The option might be to validate that the custom sp_entityid is a URI and if so use it, otherwise default to what's already there (`URL::to('auth/saml2')`).

For consistency, it might make sense to do the same as per the sp_acs, which allows you to set your own path
```
->setEntityID(URL::to($this->getConfig('sp_entityid', 'auth/saml2')))
```
 which in theory is enough in order to allow unique entity ids for different IDPs. Although for the acs to have its own domain makes sense, it is not "required" for the entity id so one might argue that this change won't really allow a custom entityid.

Reply:
Ok, happy with you just updating the docs instead of having laravel autogen :)